File: NewDDF/ddf-core/src/main/java/io/ddf2/bigquery/handlers/AggregationHandler.java
package io.ddf2.bigquery.handlers;

import io.ddf2.DDFException;
import io.ddf2.IDDF;

import java.util.List;

/**
 * Created by sangdn on 1/24/16.
 */
public class AggregationHandler implements io.ddf2.handlers.IAggregationHandler{
    @Override
    public IDDF getDDF() {
        return null;
    }

    @Override
    public double computeCorrelation(String columnA, String columnB) throws DDFException {
        return 0;
    }

    @Override
    public AggregateTypes.AggregationResult aggregate(List<AggregateTypes.AggregateField> fields) throws DDFException {
        return null;
    }

    @Override
    public AggregateTypes.AggregationResult aggregate(String query) throws DDFException {
        return null;
    }

    @Override
    public AggregateTypes.AggregationResult xtabs(List<AggregateTypes.AggregateField> fields) throws DDFException {
        return null;
    }

    @Override
    public IDDF aggregate(List<String> columns, List<String> functions) throws DDFException {
        return null;
    }

    @Override
    public double aggregateOnColumn(AggregateTypes.AggregateFunction function, String col) throws DDFException {
        return 0;
    }

    @Override
    public IDDF agg(List<String> aggregateFunctions) throws DDFException {
        return null;
    }

    @Override
    public IDDF aggregate(List<String> groupedColumns) {
        return null;
    }
}

File: spark/src/main/scala/io/ddf/spark/analytics/AggregationHandler.scala
package io.ddf.spark.analytics

import io.ddf.DDF
import io.ddf.analytics.{AggregationHandler=>CoreAggregationHandler}
import io.ddf.exception.DDFException
import java.util.{List=>JList}

import io.ddf.spark.util.SparkUtils
import io.ddf.types.AggregateTypes
import io.ddf.types.AggregateTypes.{AggregateField, AggregationResult}
import org.apache.spark.sql.AnalysisException

import scala.collection.JavaConverters._

/**
 * Created by nhanitvn on 3/21/16.
 */
class AggregationHandler(theDDF: DDF) extends CoreAggregationHandler(theDDF) {
  override def groupBy(groupedColumns: JList[String], aggregateFunctions: JList[String]): DDF = {
    // Catch SparkSQL/JSQLParser error message, parse and produce appropriate error message
    try {
      super.groupBy(groupedColumns, aggregateFunctions)
    } catch {
      case ddfException: DDFException => throw new DDFException(SparkUtils.sqlErrorToDDFError(ddfException.getMessage, buildGroupBySQL(aggregateFunctions), aggregateFunctions.asScala.toList:::groupedColumns.asScala.toList))
    }
  }

  override def aggregate (fields: JList[AggregateTypes.AggregateField]): AggregateTypes.AggregationResult = {
    // Catch SparkSQL/JSQLParser error message, parse and produce appropriate error message
    try {
      super.aggregate(fields)
    } catch {
      case ddfException: DDFException =>
        val tableName: String = this.getDDF.getTableName
        throw new DDFException(SparkUtils.sqlErrorToDDFError(ddfException.getMessage, AggregateField.toSql(fields, tableName), fields.asScala.toList.map(_.toString)))
    }
  }
}
